FlipViceFeed/app/src/main/java/com/joelimyx/flipvicefeed/DetailView/DetailActivity.java

```diff
@@ -0,0 +1,413 @@
+package com.joelimyx.flipvicefeed.detailview;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.design.widget.CollapsingToolbarLayout;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.facebook.CallbackManager;
+import com.facebook.FacebookCallback;
+import com.facebook.FacebookException;
+import com.facebook.share.Sharer;
+import com.facebook.share.model.ShareLinkContent;
+import com.facebook.share.widget.ShareButton;
+import com.facebook.share.widget.ShareDialog;
+import com.google.gson.Gson;
+import com.joelimyx.flipvicefeed.R;
+import com.joelimyx.flipvicefeed.classes.VolleySingleton;
+import com.joelimyx.flipvicefeed.detailview.adapters_holders.ArticleInfoAdapter;
+import com.joelimyx.flipvicefeed.detailview.articleobjectdata.ArticleObject;
+import com.joelimyx.flipvicefeed.detailview.articleobjectdata.Image;
+import com.joelimyx.flipvicefeed.detailview.articleobjectdata.PhotoCredit;
+import com.joelimyx.flipvicefeed.detailview.articleobjectdata.Text;
+import com.joelimyx.flipvicefeed.detailview.articleobjectdata.TextStrong;
+import com.joelimyx.flipvicefeed.detailview.articleobjectdata.Video;
+import com.joelimyx.flipvicefeed.detailview.individualarticledata.Article;
+import com.joelimyx.flipvicefeed.detailview.individualarticledata.ArticleData;
+import com.joelimyx.flipvicefeed.detailview.individualarticledata.Example;
+import com.joelimyx.flipvicefeed.detailview.individualarticledata.Gallery;
+import com.joelimyx.flipvicefeed.detailview.individualarticledata.Media;
+import com.joelimyx.flipvicefeed.main.data.ShareGsonRootObject;
+import com.joelimyx.flipvicefeed.main.data.ShareItem;
+import com.squareup.picasso.Picasso;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DetailActivity extends AppCompatActivity {
+
+    RecyclerView mRV;
+    ArticleInfoAdapter mAdapter;
+    VolleySingleton mVolleySingleton;
+    List<ArticleObject> mListOfObjectsInArticle;
+    CollapsingToolbarLayout mToolbarLayout;
+    ImageView mToolbarBackground;
+    String mToolbarBackgroundImage;
+    Toolbar mToolbar;
+    CallbackManager mCallbackManager;
+    ShareDialog mShareDialog;
+    List<ArticleObject> mMediaList;
+
+
+    public static final String TAG = "DETAIL ACTIVITY";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
Just load the detail fragment inside this activity rather than repeat all your logic in two places